pkg/ccl/changefeedccl/cdcevent/event.go

```diff
@@ -639,31 +649,35 @@ func (d *eventDecoder) DecodeKV(
 		"error decoding key %s@%s (hex_kv: %x)",
 		keys.PrettyPrint(nil, kv.Key), kv.Value.Timestamp, kvBytes))
 	log.Changefeed.Errorf(ctx, "terminal error decoding KV: %v", err)
-	return Row{}, err
+	return Row{}, DecodeOK, err
 }
 
 // decodeKV decodes key value at specified schema timestamp.
 func (d *eventDecoder) decodeKV(
 	ctx context.Context, kv roachpb.KeyValue, rt RowType, schemaTS hlc.Timestamp, keyOnly bool,
-) (Row, error) {
-	if err := d.initForKey(ctx, kv.Key, schemaTS, keyOnly); err != nil {
-		return Row{}, err
+) (Row, DecodeStatus, error) {
+	status, err := d.initForKey(ctx, kv.Key, schemaTS, keyOnly)
+	if err != nil {
```

Comment:
Won't this catch the `Row{}, DecodeSkip.., nil` case too? in which case we don't need the following if stmt.